storage/engine/rocksdb.go

```diff
@@ -166,6 +167,40 @@ func (r *RocksDB) getInternal(key proto.EncodedKey, snapshotHandle *C.DBSnapshot
 	return cStringToGoBytes(result), nil
 }
 
+// GetProto fetches the value at the specified key and unmarshals it.
+func (r *RocksDB) GetProto(key proto.EncodedKey, msg gogoproto.Message) (
+	ok bool, keyBytes, valBytes int64, err error) {
+	return r.getProtoInternal(key, msg, nil)
+}
+
+func (r *RocksDB) getProtoInternal(key proto.EncodedKey, msg gogoproto.Message,
+	snapshotHandle *C.DBSnapshot) (ok bool, keyBytes, valBytes int64, err error) {
+	if len(key) == 0 {
+		err = emptyKeyError()
+		return
+	}
+	var result C.DBString
+	if err = statusToError(C.DBGet(r.rdb, snapshotHandle, goToCSlice(key), &result)); err != nil {
+		return
+	}
+	if result.len <= 0 {
+		return
+	}
+	ok = true
+	if msg != nil {
+		// Make a byte slice that is backed by result.data. This slice
+		// cannot live past the lifetime of this method, but we're only
+		// using it to unmarshal the proto.
+		// data := C.GoBytes(unsafe.Pointer(result.data), result.len)
```

Comment:
Either explain why this commented-out line is here or remove it.

The assumption that Unmarshal always copies any bytes it uses from the input data scares me. It should generally be true since Unmarshal doesn't want to keep the input data from getting GC'd, but this greatly increases the stakes for any code paths that simply slice bytes out of the input without copying. I'm not sure the performance improvements here are worth the loss of memory safety.